judge/views/problem.py

```diff
@@ -490,7 +492,9 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         context = super(ProblemList, self).get_context_data(**kwargs)
         context['hide_solved'] = 0 if self.in_contest else int(self.hide_solved)
-        context['show_types'] = 0 if self.in_contest else int(self.show_types)
```

Comment:
why did you avoid
```py
context['show_types'] = int(not self.contest.hide_problem_tags) if self.in_contest else int(self.show_types)
```

Reply:
> why did you avoid
> 
> ```python
> context['show_types'] = int(not self.contest.hide_problem_tags) if self.in_contest else int(self.show_types)
> ```

I wasn't sure whether or not the self.show_types attribute was used elsewhere and had to be changed as well

Reply:
`get_context_data` doesn't look like the right place to modify `self.show_types`. what happens if you put your changes in `setup_problem_list`?